File: intro-dev-tsjs-2024-03-28/3.ts
import ollama from 'ollama';

const prompt = "why is the sky blue";

const output = await ollama.generate({ model: "llama2", prompt: prompt, stream: true })

console.log(output)

// for await (const part of output) {
  // console.log(part)
//   process.stdout.write(part.response)
//   if (part.done === true) {
//     console.log(`\nstats: ${(part.eval_count / (part.eval_duration / 1000000000)).toFixed(2)} tokens per second`)  }
// }

File: 2024-04-01-intro-dev-python/3.py
import ollama

prompt = "why is the sky blue";

output = ollama.generate(model="llama2", prompt=prompt, stream=True)

# print(output)

for chunk in output:
    print(chunk["response"], end='', flush=True)
    if chunk["done"] == True:
        tokens_per_second = chunk["eval_count"] / (chunk["eval_duration"] / 1000000000)
        print("\nstats: {:.2f} tokens per second".format(tokens_per_second))

